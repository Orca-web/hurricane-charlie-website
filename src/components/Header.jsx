import React, { Component } from 'react'

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hamburgerOpen: false
    }
    this.hamburgerClick = this.hamburgerClick.bind(this);
  }


  hamburgerClick() {
    this.setState({hamburgerOpen: !this.state.hamburgerOpen});
    
    // nav.classList.toggle('hidden');
  }

  render() {
    let hamburgerClass = `hamburger hamburger--spin ham-handle ${this.state.hamburgerOpen ? 'is-active' : ''}`;

    return (
      <header>
        <img src="assets/img/hclogo.png" alt="Hurricane Charlie Logo" />
        <nav className={this.props.width > 639 ? "" : "hidden"}>
          <li><a href="#">gallery</a></li>
          <li><a href="#">contact</a></li>
          <li><a href="#">shop</a></li>
        </nav>
        <div onClick={this.hamburgerClick} className="hamburger-container">
          <button className={hamburgerClass}>
            <span className="hamburger-box">
              <span className="hamburger-inner"></span>
            </span>
          </button>
        </div>
      </header>
    )
  }
}

export default Header