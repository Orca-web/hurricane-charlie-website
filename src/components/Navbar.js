import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class Navbar extends Component {
  render() {
    const { burgerOpen, navData } = this.props;
    const nav = navData.map(item => {
      const navClickedStyle = {
        borderBottom: item.active && !burgerOpen ? "3px solid #FF2E63" : "none",
        color: item.active && burgerOpen ? "#FF2E63" : "black"
      };
      return (
        // <li 
        //   id={item.name} key={item.id} 
        //   onClick={this.props.navClick} 
        //   style={navClickedStyle} 
        // >{item.name}</li>
        <NavLink key={item.id} to={item.url}>{item.name}</NavLink>
      );
    });

    // <NavLink to="/">Home</NavLink>
  
    return(
      <nav>
        { nav }
      </nav>
    )
  }
}

export default Navbar;