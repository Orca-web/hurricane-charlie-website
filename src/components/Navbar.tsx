import React from 'react';
import { NavLink } from 'react-router-dom';
import { NavData } from '../globals/navData';

const activeNav = { borderBottom: "3px solid #FF2E63" };

interface NavbarProps {
  navData: NavData[];
}

const Navbar = ({
  navData
}: NavbarProps) => (
    <nav>
      {navData.map(item => (
        <li key={item.id}>
          <NavLink to={item.url} exact activeStyle={activeNav}>
            {item.name}
          </NavLink>
        </li>
      ))}
    </nav>
);

export default Navbar;