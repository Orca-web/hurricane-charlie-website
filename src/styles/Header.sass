header
  display: grid
  position: relative

  img
    height: 50px
    justify-self: center
    padding: 15px

  nav
    padding: 10px 0
    justify-self: center

    li
      display: block
      text-align: center
      padding: 10px

  .hamburger-container
    position: absolute
    right: 15px
    top: 40px
    transform: translateY(-50%)

@import './hamburger'

@media screen and (min-width: 640px)

  header
    grid-template-columns: 70px 1fr

    img
      padding: 10px

    nav
      padding: 0
      align-self: center
      justify-self: right

      li
        display: inline-block