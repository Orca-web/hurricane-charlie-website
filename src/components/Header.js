import React, { Component } from 'react';
import logo from '../assets/img/hclogo.png';
import Navbar from './Navbar';

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = { burgerOpen: false, width: 0 }
    this.updateWindowSize = this.updateWindowSize.bind(this);
    this.burgerClick = this.burgerClick.bind(this);
  }

  burgerClick() {
    this.setState({ burgerOpen: !this.state.burgerOpen });
  }

  componentDidMount() {
    this.updateWindowSize();
    window.addEventListener('resize', this.updateWindowSize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowSize);
  }

  updateWindowSize() {
    this.setState({ width: window.innerWidth })
  }

  render() {
    const { burgerOpen, width } = this.state;
    const burgerClass = `hamburger hamburger-spin ${burgerOpen ? 'is-active' : ''}`;

    const burger = width < 640 ? (
      <div onClick={this.burgerClick} className="hamburger-container">
        <div className={burgerClass}>
          <span className="hamburger-box">
            <span className="hamburger-inner"></span>
          </span>
        </div>
      </div>
    ) : null;

    const nav = width > 640 || burgerOpen ? (
      <Navbar 
        navClick={this.props.navClick} 
        navData={this.props.navData}
        burgerOpen={burgerOpen}
      />
    ) : null;
    
    return (
      <header>
        <img src={logo} alt="Hurricane Charlie Logo" />
        {nav}
        {burger}
      </header>
    )
  }
}

export default Header