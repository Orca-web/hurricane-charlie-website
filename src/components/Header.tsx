import React, { Component } from 'react';
import logo from '../images/hclogo.png';
import Navbar from './Navbar';
import navData from '../globals/navData';

class Header extends Component {
  state = {
    burgerOpen: false,
    width: 0
  }

  burgerClick = () => {
    this.setState({ burgerOpen: !this.state.burgerOpen });
  }

  componentDidMount() {
    this.updateWindowSize();
    window.addEventListener('resize', this.updateWindowSize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowSize);
  }

  updateWindowSize = () => {
    this.setState({ width: window.innerWidth })
  }

  render() {
    const { burgerOpen, width } = this.state;
    const burgerClass = `hamburger hamburger-spin ${burgerOpen ? 'is-active' : ''}`;
    
    return (
      <header>
        <img src={logo} alt="Hurricane Charlie Logo" />

        {width < 640 && (
          <div onClick={this.burgerClick} className="hamburger-container">
            <div className={burgerClass}>
              <span className="hamburger-box">
                <span className="hamburger-inner"></span>
              </span>
            </div>
          </div>
        )}

        {(width > 640 || burgerOpen) && <Navbar navData={navData} />}
      </header>
    )
  }
}

export default Header