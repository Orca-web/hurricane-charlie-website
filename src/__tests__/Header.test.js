import React from "react";
import { shallow, mount } from "enzyme";
import Header from "../components/Header";
import Navbar from "../components/Navbar";

const minProps = { navData: [{name: test, id: 0}] };

const app = shallow(<Header {...minProps} />);

describe("Header", () => {
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });

  describe("screen at full width", () => {
    it("nav showing", () => {
      expect(app.find(Navbar).length).toBe(1);
    });

    it("hamburger hidden", () => {
      expect(app.find('.hamburger-container').exists()).toEqual(false)
    });
  });
  
  describe("screen at mobile width", () => {  
    global.innerWidth = 600;
    global.dispatchEvent(new Event('resize'));

    const wrapper = shallow(<Header {...minProps} />);

    it("nav hidden", () => {
      expect(wrapper.find(Navbar).length).toBe(0);
    });

    it("hamburger showing", () => {
      expect(wrapper.find('.hamburger-container').exists()).toEqual(true);
    });

    it("hamburger clicked nav showing", () => {
      wrapper.find('.hamburger-container').simulate('click');
      expect(wrapper.find(Navbar).length).toBe(1);
    });
  });

});
