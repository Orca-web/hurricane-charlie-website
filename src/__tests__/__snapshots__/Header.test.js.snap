// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    navData={
      Array [
        Object {
          "id": 0,
          "name": [Function],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Hurricane Charlie Logo"
          src="assets/img/hclogo.png"
        />,
        <Navbar
          navData={
            Array [
              Object {
                "id": 0,
                "name": [Function],
              },
            ]
          }
        />,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Hurricane Charlie Logo",
          "src": "assets/img/hclogo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "navData": Array [
            Object {
              "id": 0,
              "name": [Function],
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Hurricane Charlie Logo"
            src="assets/img/hclogo.png"
          />,
          <Navbar
            navData={
              Array [
                Object {
                  "id": 0,
                  "name": [Function],
                },
              ]
            }
          />,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Hurricane Charlie Logo",
            "src": "assets/img/hclogo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navData": Array [
              Object {
                "id": 0,
                "name": [Function],
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
